### PR TITLE
Bugfix/pp9 9701 thumbs share dialog

### PR DESCRIPTION
**Pull request by rjs-picturepark**
New changes were needed on this issue has the previous solution broke backwards compatibility. Using the old solution creates a lot of unnecessary request and possibly breaks [PP9-10247 - Create a Union type for the ContentDownloadDialogOptions content], to be checked with Tiago.

### CODE REVIEW

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ import {
 import { BasketService } from '../../shared-module/services/basket/basket.service';
 import { ContentDownloadDialogService } from '../content-download-dialog/content-download-dialog.service';
 import { ContentModel } from '../../shared-module/models/content-model';
+import { Observable } from 'rxjs';
+import { ContentService } from '@picturepark/sdk-v1-angular';
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts]: This one should be included in the import at line 5.

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ import {
 import { BasketService } from '../../shared-module/services/basket/basket.service';
 import { ContentDownloadDialogService } from '../content-download-dialog/content-download-dialog.service';
 import { ContentModel } from '../../shared-module/models/content-model';
+import { Observable } from 'rxjs';
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts]: Why not move this import below the MatDialog import at line 2?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts [hunk 3]
```diff
@@ -23,45 +32,51 @@ import { ContentModel } from '../../shared-module/models/content-model';
...
     private basketService: BasketService,
+    private contentService: ContentService,
     private contentDownloadDialogService: ContentDownloadDialogService,
     public dialog: MatDialog,
   ) {
 
     super();
 
-    const basketSubscription = this.basketService.basketChange.subscribe((items) => this.basketItems = items);
+    const basketSubscription = this.basketService.basketChange.subscribe(async (items) => {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts]: Remove async

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts [hunk 4]
```diff
@@ -76,4 +91,18 @@ export class BasketComponent extends BaseComponent implements OnInit {
     this.configActions = this.pictureParkUIConfig['BasketComponent'];
   }
 
+  private fetchContents(): Observable<ISearchResult<Content>> {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/basket/basket.component.ts]: Where it's being used close the subscription

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts [hunk 5]
```diff
@@ -95,4 +114,18 @@ export class ContentItemThumbnailComponent extends BaseBrowserItemComponent<Cont
     event.path[0].src = BROKEN_IMAGE_URL;
   }
 
+  private fetchContentById(id: string): Observable<ISearchResult<Content>> {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts]: Extract this fetch method and the fetch in the basket component ts, create file in lib in angular projects (think about a name)

**Pull request merged**